### ISSUE TITLE
Pitch/Yaw ignored when spawning display entities

### ISSUE THREAD

**Issue by tal5** [PaperSpigot/Paper#9418]
### Expected behavior

For pitch and yaw to be used when spawning, as with other entities.

### Observed/Actual behavior

When spawning a display entity, pitch and yaw seem to be ignored and always become `0`.

### Steps/models to reproduce

```java
Location location = player.getEyeLocation();
getLogger().info(location.toString());
Display display = player.getWorld().spawn(location, BlockDisplay.class);
getLogger().info(display.getLocation().toString());
getLogger().info(String.valueOf(display.getLocation().equals(location)));
```
Running the above code will log the first location with the player's yaw and pitch, then the display's location with `0` yaw and pitch, then show that they're not equal:
```
[23:44:07 INFO]: [PaperTestPlugin] Location{world=CraftWorld{name=world},x=-108.5082338140295,y=101.99942319657657,z=354.9048698050415,pitch=90.0,yaw=162.454}
[23:44:07 INFO]: [PaperTestPlugin] Location{world=CraftWorld{name=world},x=-108.5082338140295,y=101.99942319657657,z=354.9048698050415,pitch=0.0,yaw=0.0}
[23:44:07 INFO]: [PaperTestPlugin] false
```

### Plugin and Datapack List

```
> plugins
[23:54:02 INFO]: Server Plugins (1):
[23:54:02 INFO]: Bukkit Plugins:
[23:54:02 INFO]:  - PaperTestPlugin
```
```
> datapack list
[23:54:21 INFO]: There are 2 data pack(s) enabled: [vanilla (built-in)], [file/bukkit (world)]
[23:54:21 INFO]: There are no more data packs available
````

### Paper version

```
> version
[00:00:49 INFO]: Checking version, please wait...
[00:00:50 INFO]: This server is running Paper version git-Paper-56 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: 2c9c5e4)
You are running the latest version
Previous version: git-Paper-54 (MC: 1.20.1)
```

### Other

I understand that generally you'd use the built-in transformation and not the base yaw/pitch, but I believe there's still a benefit to fixing this if possible (e.g. custom entity systems with generic logic and whatnot).
Especially seeing as this works fine in vanilla commands: `/summon block_display ~ ~ ~ {Rotation:[50F,50F],block_state:{Name:"minecraft:grass_block"}}`

**Comment by Owen1212055** [comment]
Related to https://github.com/PaperMC/Paper/pull/5462
In general, I think the spawning logic bukkit has should be cleaned up because there's no reason why pitch/yaw should be excluded like this.

**Comment by Black-Hole** [comment]
Should be fixed by upstream and included in Paper with 2fa8efce9b49bc060e3d332b47f454dadc70188e

**Comment by Doc94** [comment]
> Should be fixed by upstream and included in Paper with [2fa8efc](https://github.com/PaperMC/Paper/commit/2fa8efce9b49bc060e3d332b47f454dadc70188e)

i think you mean [this](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/a2fafdd1d2f0f70f378dfda714147c2a202c4c16#nms-patches/net/minecraft/server/level/EntityTrackerEntry.patch)? that was for only Player rotation not?

**Comment by Darwin1546** [comment]
It's still present in Paper-117
